middleware/event_stream.go

```diff
@@ -1,46 +1,26 @@
 package middleware
 
 import (
-	"encoding/json"
-
-	"github.com/RedHatInsights/sources-api-go/internal/events"
-	"github.com/RedHatInsights/sources-api-go/kafka"
 	l "github.com/RedHatInsights/sources-api-go/logger"
-	"github.com/RedHatInsights/sources-api-go/util"
+	"github.com/RedHatInsights/sources-api-go/model"
+	"github.com/RedHatInsights/sources-api-go/service"
 	"github.com/labstack/echo/v4"
 )
 
-// producer instance used to send messages - default just an empty instance of
-// the struct.
-var producer = events.EventStreamProducer{Sender: &events.EventStreamSender{}}
-
-/*
-	Function that takes an event-type as a string and then returns a middleware
-	function that raises the specified event if the handler operation succeeds.
-
-	It bails out if there isn't a `resource` field on the context which should
-	be a model.ToEvent() call in the handler.
-*/
-func RaiseEvent(next echo.HandlerFunc) echo.HandlerFunc {
+// RaiseEventMiddleware calls the "RaiseEvent" function once the previous handler has succeeded. It grabs the resource
+// and the event type from the context.
+func RaiseEventMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
 	return func(c echo.Context) error {
 		// first call the handler function (or the next middlware)
 		err := next(c)
 		if err != nil {
 			return err
 		}
 
-		// specifically skip raising an event if this is set - usually when
-		// a create action happened but we do not want to re-raise the
-		// event.
-		if c.Get("skip_raise") != nil {
-			l.Log.Infof("skipping raise event per skip_raise set on context")
-			return nil
-		}
-
```

Comment:
So this part needs to stay - I am surprised my tests didn't catch it.

Reply:
I think it is because of the following:

```go
	f := raiseMiddleware(func(c echo.Context) error {
		c.Set("skip_raise", struct{}{})
		return c.NoContent(http.StatusNoContent)
	})
```

It seems to be returning the `StatusNoContent` all the time, which causes the tests to pass no matter what.